YogaKit/YGLayout.m

```diff
@@ -318,6 +312,14 @@ static void YGAttachNodesFromViewHierachy(UIView *const view)
   if (yoga.isLeaf) {
     YGRemoveAllChildren(node);
     YGNodeSetMeasureFunc(node, YGMeasureView);
+
+    YGSize oldSize = yoga.cachedSize;
+    YGSize newSize = YGMeasureView(node, 0, YGMeasureModeUndefined, 0, YGMeasureModeUndefined);
```

Comment:
A few points:

1. My benchmarks show me that this should not be a concern. `UIImageView` `sizeThatFits` seems instantaneous because it simply returns the size of the image. `UILabel`'s `sizeThatFits` takes 0.0000567 seconds (5% STD) on my iPhone 6S.

2. I'm not quite sure I understand your point. Could you explain again? :)

I understand the drawbacks you are putting forward. But, having used this auto-detection in the app I'm currently working, I can tell you it's a lifesaver. I tried for a while to mark views as dirty when necessary, but it's very bug-prone. If we are really worried about performance, perhaps we could make it an optional feature?

```
[YGYoga setDirtyNodeDetectionEnabled:YES];
```